### PR TITLE
Add Formik to example application

### LINKED ISSUE #888: Add Formik to example application
Repeat #887 but for Formik

### PR DESCRIPTION
**Pull request by penx**
Fixes #888

### CODE REVIEW

File: packages/example-application/src/formik.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+import { Formik, Form, Field as FormikField } from 'formik';
+
+const Field = ({ component: Component, ...props }) => (
+  <FormikField {...props}>{({ field, meta }) => <Component {...props} input={field} meta={meta} />}</FormikField>
+);
+
+const Checkbox = ({ input, ...props }) => <GovUK.Checkbox {...input} {...props} />; //eslint-disable-line
+const DateField = ({ meta, input: { onChange, onBlur, ...input }, ...props }) => (
+  <GovUK.DateField
```
Comment (penx) [inline packages/example-application/src/formik.tsx:13]: this is a bit of a mess, the DateField in particular, though the point of this is to illustrate issues with the component API that need to be addressed in 0.10

File: packages/example-application/src/formik.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                      validate={(value) => (value ? undefined : 'Please select an animal')}
+                    >
+                      <option value="cat">Cat</option>
+                      <option value="other-feline">Other feline</option>
+                      <option value="other-non-feline">Other non feline</option>
+                    </Field>
+                    {/*
+                    TODO: need to be able to pass props to file input
+                    https://github.com/final-form/react-final-form/issues/663
+                    */}
```
Comment (penx) [inline packages/example-application/src/formik.tsx:166]: https://github.com/govuk-react/govuk-react/issues/900

File: packages/example-application/src/index.spec.tsx [hunk 3]
```diff
@@ -10,6 +10,37 @@ describe('When a user loads the application,', () => {
     beforeEach(() => {
       cy.get('a').contains('Start now').click();
     });
+    describe('clicks submit,', () => {
+      beforeEach(() => {
+        cy.get('button').contains('Submit').click();
+      });
+      it('should show error messages', () => {
+        cy.contains('Error summary').should('be.visible');
+        // TODO: test that error summary items are anchor links to relevant field
```
Comment (penx) [inline packages/example-application/src/index.spec.tsx:19]: https://github.com/govuk-react/govuk-react/issues/480

File: packages/example-application/src/index.spec.tsx [hunk 4]
```diff
@@ -42,6 +73,100 @@ describe('When a user loads the application,', () => {
...
+        cy.contains('Error summary').parent().contains('Please answer the question').should('be.visible');
+
+        cy.get('[name="firstName"]').parent().contains('Please enter a first name').should('be.visible');
+        cy.get('[name="description"]').parent().contains('Please enter a description').should('be.visible');
+        cy.get('[name="nationality"]')
+          .parent()
+          .parent()
+          .contains('Please select at least one nationality')
+          .should('be.visible');
+        // TODO: better selector for date of birth
```
Comment (penx) [inline packages/example-application/src/index.spec.tsx:160]: this is low priority but something to think about when revisiting this file

File: packages/example-application/src/formik.tsx (hunk 1 above)
Comment (marksy) [reply packages/example-application/src/formik.tsx:15]: Maybe create an issue to improve?

**Pull request merged**